src/RevProc.cc

```diff
@@ -1638,15 +1638,32 @@ unsigned RevProc::GetNextHartToDecodeID() const {
 
 void RevProc::MarkLoadComplete(const MemReq& req){
 
-  auto it = LSQueue->find(req.LSQHash());
-  if( it != LSQueue->end()){
-    DependencyClear(it->second.Hart, it->second.DestReg, (it->second.RegType == RevRegClass::RegFLOAT));
-    LSQueue->erase(it);
+  auto it = LSQueue->equal_range(req.LSQHash());      // Find all outstanding dependencies for this register
+  bool addrMatch = false;
+  if( it.first != LSQueue->end()){                  
+    for (auto i = it.first; i != it.second; ++i){    // Iterate over all outstanding loads for this reg (if any)
+        if(i->second.Addr == req.Addr){
+          if(LSQueue->count(req.LSQHash()) == 1){   // Only clear the dependency if this is the LAST outstanding load for this register
+            DependencyClear(i->second.Hart, i->second.DestReg, (i->second.RegType == RevRegClass::RegFLOAT));
+          }
+          MemReq temp = req;
+          sfetch->MarkInstructionLoadComplete(temp);
```

Comment:
Why do we need to create a `temp` and send a non-const reference to it? All the other uses of MarkLoadComplete use `const MemReq&` -- see https://github.com/tactcomplabs/rev/pull/219

> Use `std::function<void(const MemReq&)>` consistently throughout the code. Previously `std::function<void(MemReq)>` was used in some places. While it may work because of conversions, it is not as clean.

Reply:
At one point there was an issue w/ a what appeared to be a double-delete of req when it was being erased from multiple functions. This appears to no longer be necessary. Updated to just use req